internal/http/services/owncloud/ocdav/locks.go

```diff
@@ -190,11 +191,18 @@ func (cls *cs3LS) Create(ctx context.Context, now time.Time, details LockDetails
 	// see: http://www.webdav.org/specs/rfc2518.html#n-lock-tokens
 	token := uuid.New()
 
+	u := ctxpkg.ContextMustGetUser(ctx)
+
+	// add metadata via opaque
+	o := utils.AppendPlainToOpaque(nil, "lockownername", u.GetDisplayName())
```

Comment:
Here you are persisting the displayname ... but that might change over time. The lock owner is already persisted by id.

Either the web ui has to look up the displayname whan showing the lock or the server has to look up the user displayname when trying to render the lock. 

```suggestion
```

Reply:
That would mean 
A) The server needs to make another call on propfinds. Possibly multiple when there are different files locked.
B) The server needs to allow `GetUserByID` for non admin users. I would try to avoid this as it is leaking data.

Locks are usually short lived, so the chance the username changes while the lock is active is very low. And even if, it wouldn't bring much confusion. Hence I was going for the simplest solution.

Would you prefer option A?

Reply:
There should already be a cache for A) as propfinds already have to fetch user data to fill the `oc:owner-display-name` property 👀 oh well it does not do that:
```
sublog.Debug().Msg("TODO fetch user displayname")
```

maybe we don't even need that in web any longer as the owner/manager is part of the space ... It is only in the code for web ...

B) How is a `GetUserByID` call leaking when you can already search for sharing recipients? Listing shares also has the problvem that we have to fetch user metadata data for every share. Another endpoint where I would like to omit the metadata. If you already have the userid I think it is ok to allow users to fetch the displayname, username and avatar. the same data you can fetch via the share recipients lookup.

We could even maintain a list of users the currently logged in user has recently interacted with. On the server as well as in the browser... at some point in the future...

For now, I'd go with A) as it will only cause a user lookup when a resource is locked.

Reply:
You search for recipients by name. That means you know the name already. If you just know the ID you can't get any more information on this user as of now. This is good because it is secure. But we can discuss this in the future if you want.

Are you really SURE we want to have the extra load on the server just to avoid a username that is outdated in an edge case for not more than 1 Minute? It will make the code more complicated, the logic more complicated and the put extra pressure on (already slow) propfind calls. 

I can't see any benefit in that.